### ISSUE TITLE
Bug: [Windows gitbash sdk current java error: Not using any version of java]

### ISSUE THREAD

**Issue by mbstockm** [sdkman/sdkman-cli#1075]
**Bug report**

I receive the error "Not using any version of java" when running sdk current java on windows 11 using gitbash.  

Michael@Michael-Desktop MINGW64 /
$ sdk current java

Not using any version of java

When running sdk list java all of the versions are displayed, and all of the versions I have installed. If I use the sdk use command it successfully changes  the java version, and then when I run sdk current java the version is displayed. I thought the default version was also supposed to display by running sdk current java. 

Michael@Michael-Desktop MINGW64 /
$ sdk use java 11.0.12-open

Using java version 11.0.12-open in this shell.

Michael@Michael-Desktop MINGW64 /
$ sdk list java

Michael@Michael-Desktop MINGW64 /
$ sdk current java

Using java version 11.0.12-open

**To reproduce**

Install gitbash for windows, on windows 11 following instruction on sdkman site. Install multiple versions of java, and then select one as the default version. Try running sdk current java and error should reproduce. 

**System info**

Windows 11 Pro 2000 Build 22000

$ bash --version
GNU bash, version 4.4.23(1)-release (x86_64-pc-msys)
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Michael@Michael-Desktop MINGW64 /
$ sdk version

SDKMAN 5.14.1

**Comment by mbstockm** [comment]
I should also mention, that the java version is changed under the candidates current folder on Windows. So it is working correctly on the system, just not displaying as you would expect within gitbash sdk commands. 

C:\Users\Michael\.sdkman\candidates\java\current does change between versions.

**Comment by SvenGasterstaedtGebit** [comment]
Note that this probably also breaks the auto_env.

As the current list of candidate (and default is empty) sdk man fails to restore the default version.

![image](https://user-images.githubusercontent.com/131267808/233079003-653899d1-2b91-47d1-943a-82d57b1ec3fb.png)
![image](https://user-images.githubusercontent.com/131267808/233079305-26942b75-51e7-43ae-ad53-c2da30311e3b.png)

**Comment by CodeGetters** [comment]
I have the same problem. Have you solved it yet？😥

**Comment by mbstockm** [comment]
This is still an issue for me.  I just tried it again. 

Installed JDK  11.0.19-amzn.

Then tried setting it as default. 

mbstockm@17926WLS MINGW64 ~
$ sdk default java 11.0.19-amzn

Default java version set to 11.0.19-amzn

Then tried checking current java

mbstockm@17926WLS MINGW64 ~
$ sdk current java

Not using any version of java

However it did still actually change my Java version in windows, since my JAVA_HOME looks at the candidates current, and that is changed by this command. It just isn't showing when sdk list java is run.

**Comment by CodeGetters** [comment]
yeah, it also changed my version of Java on windows, but only in git-MINGW, in the global environment my java environment still shows the version of java I installed before sdkman. Later, I found that I can use sdkman to install java in CentOS.
So I suspect that the java environment before I installed it is in conflict with the java in sdkman or other reason

**Comment by randeexiong** [comment]
问题目前依旧存在，没有解决
